testutils/clustermanager/util_test.go

```diff
@@ -35,15 +38,20 @@ func TestGetResourceName(t *testing.T) {
 		// restore GetOSEnv
 		common.GetOSEnv = oldFunc
 	}()
-	common.GetOSEnv = func(key string) string {
-		return buildNumStr
-	}
-
-	out, err := getResourceName(ClusterResource)
-	if nil != err {
-		t.Fatalf("getting resource name for cluster, wanted: 'no error', got: '%v'", err)
-	}
-	if out != expOut {
-		t.Fatalf("getting resource name for cluster, wanted: '%s', got: '%s'", expOut, out)
+	for _, data := range datas {
+		common.GetOSEnv = oldFunc
```

Comment:
Actually you can move this out the loop and use the mock in all cases.

/approve
/lgtm
/hold

Holding in case you want to address the comment.

Reply:
Sorry, spoke too soon. Actually, your mock function should correctly handle the two expected keys, PROW_JOB_ID and BUILD_ID (and because it changes according to the data, it must be in the loop, but as a simple assignment).

Reply:
done